src/hevm/src/EVM/Types.hs

```diff
@@ -45,35 +50,69 @@ import qualified Text.Read
 -- Some stuff for "generic programming", needed to create Word512
 import Data.Data
 
--- We need a 512-bit word for doing ADDMOD and MULMOD with full precision.
-mkUnpackedDoubleWord "Word512" ''Word256 "Int512" ''Int256 ''Word256
-  [''Typeable, ''Data, ''Generic]
+
+data Buffer
+  = ConcreteBuffer ByteString
+  | SymbolicBuffer [SWord 8]
 
 newtype W256 = W256 Word256
   deriving
     ( Num, Integral, Real, Ord, Enum, Eq
     , Bits, FiniteBits, Bounded, Generic
     )
 
--- | convert between (WordN 256) and Word256
-type family ToSizzle (t :: Type) :: Type where
-    ToSizzle W256 = (WordN 256)
-    ToSizzle Addr = (WordN 160)
+data Word = C Whiff W256 --maybe to remove completely in the future
+
+instance Show Word where
+  show (C _ x) = show x
+
+-- | Symbolic words of 256 bits, possibly annotated with additional
+--   "insightful" information
+data SymWord = S Whiff (SWord 256)
+
+instance Show SymWord where
```

Comment:
can you move the other instances of `SymWord` to here as well? Right now we are getting a lot of warnings about orphan instances